test/no-incorrect-deep-equal.js

```diff
@@ -0,0 +1,192 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-incorrect-deep-equal';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errorLiteral = {
+	ruleId: 'no-incorrect-deep-equal',
+	messageId: 'no-deep-equal-with-literal'
+};
+
+const errorTemplate = {
+	ruleId: 'no-incorrect-deep-equal',
+	messageId: 'no-deep-equal-with-template'
+};
+
+const errorUndefined = {
+	ruleId: 'no-incorrect-deep-equal',
+	messageId: 'no-deep-equal-with-undefined'
+};
+
+const header = 'const test = require(\'ava\');\n';
+
+ruleTester.run('no-incorrect-deep-equal', rule, {
+	valid: [
+		`
+			${header}
+			test(t => {
+				t.deepEqual(expression, otherExpression);
+			});
+		`,
+		`
+			${header}
+			test(t => {
+				t.deepEqual(expression, {});
+			});
+		`,
+		`
+			${header}
+			test(t => {
+				t.deepEqual(expression, []);
+			});
+		`,
+		`
+			${header}
+			test(t => {
+				t.notDeepEqual(expression, []);
+			});
+		`
+	],
+	invalid: [
+		{
+			code: `
+				${header}
+				test(t => {
```

Comment:
I know this doesn't matter here, but AVA requires tests to have a title, so would be good to do that. Can just be `test('x', ..`.